Lib/unittest/mock.py

```diff
@@ -2693,70 +2749,39 @@ def mock_open(mock=None, read_data=''):
 
     `read_data` is a string for the `read`, `readline` and `readlines` of the
     file handle to return.  This is an empty string by default.
-    """
-    _read_data = _to_stream(read_data)
-    _state = [_read_data, None]
 
-    def _readlines_side_effect(*args, **kwargs):
-        if handle.readlines.return_value is not None:
-            return handle.readlines.return_value
-        return _state[0].readlines(*args, **kwargs)
-
-    def _read_side_effect(*args, **kwargs):
-        if handle.read.return_value is not None:
-            return handle.read.return_value
-        return _state[0].read(*args, **kwargs)
-
-    def _readline_side_effect(*args, **kwargs):
-        yield from _iter_side_effect()
-        while True:
-            yield _state[0].readline(*args, **kwargs)
-
-    def _iter_side_effect():
-        if handle.readline.return_value is not None:
-            while True:
-                yield handle.readline.return_value
-        for line in _state[0]:
-            yield line
-
-    def _next_side_effect():
-        if handle.readline.return_value is not None:
-            return handle.readline.return_value
-        return next(_state[0])
+    `data` is any value acceptable for `dict` constructor. The keys
+    represent file names. If a file name, found in `data`, is
+    mock-opened, and the value matching the file name in `data` will
```

Comment:
This sentence doesn't read right to me. Should the "and" be replaced with "then" here?